main.go

```diff
@@ -35,100 +33,93 @@ import (
 	"sync"
 	"syscall"
 	"unicode/utf8"
+
+	"github.com/satori/go.uuid"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+const (
+	dirMode = "dir"
+	dnsMode = "dns"
 )
 
-// A single result which comes from an individual web
-// request.
-type Result struct {
+// result is the response of an individual web request
+type result struct {
```

Comment:
since nothing here is exported for other packages to use we can use lower case names